File: Demo/ObjC/TinyConsoleObjCDemo/TinyConsoleObjCDemo/DemoController.m
//
//  DemoController.m
//  TinyConsoleObjCDemo
//
//  Created by Nash Zhou on 3/5/18.
//  Copyright © 2018 nashzhou. All rights reserved.
//

#import "DemoController.h"

#if DEBUG
#import <TinyConsolePlugin/TinyConsole.h>
#endif

@implementation DemoController

-(void)viewDidLoad
{
    [super viewDidLoad];
    self.title = @"Demo";
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
}

-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 30;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath;
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.textLabel.text = [NSString stringWithFormat:@"Row %ld", (long)indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
#if DEBUG
    [TinyConsole printText:[NSString stringWithFormat:@"Tapped on %ld", (long)indexPath.row]];
#endif
}

@end

File: Demo/Swift/TinyConsoleSwiftDemo/TinyConsoleSwiftDemo/DemoController.swift
//
//  DemoController.swift
//  TinyConsoleSwiftDemo
//
//  Created by Nash Zhou on 3/5/18.
//  Copyright © 2018 nashzhou. All rights reserved.
//

import Foundation
import UIKit

import TinyConsolePlugin.TinyConsole

class DemoController: UITableViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Demo"
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
    }
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 30
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
        cell.textLabel?.text = "Row \(indexPath.row)"
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        TinyConsole.printText("Tapped on \(indexPath.row)")
    }
    
}
